core/src/test/scala/com/github/gpoirier/barrage/GameStateSpec.scala

```diff
@@ -49,6 +49,48 @@ class GameStateSpec extends AnyFlatSpec with Matchers {
     op.run(before).fold(fail(_), _._2 shouldBe "No empty action spot left (Two)")
   }
 
+  it should "ensure spin costs are correct" in {
+    val initial = GameState.initial(NonEmptyList.of(USA, Italy, Germany))
+    val addCredits = (lens.currentPlayerState composeLens lens.playerCredits).modify(_ ++ 14.credits)
+    val addEng = (lens.currentPlayerState composeLens lens.engineers).modify(_ ++ EngineerCount(1))
+    val before = (addCredits compose addEng)(initial)
+
+    val getEng: StateM[GameState, EngineerCount] = StateT.inspect((lens.currentPlayerState composeLens lens.engineers).get)
+    val getCredits: StateM[GameState, Credits] = StateT.inspect((lens.currentPlayerState composeLens lens.playerCredits).get)
+
+    val op = for {
+      _ <- getEng.map(_ shouldBe 13.eng)
```

Comment:
This test will fail if we implement invariants for things like the engineer's count that can never be more than 12